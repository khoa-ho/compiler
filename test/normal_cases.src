2 + 3;
2 - 3;
2 * 3;
2 / 3;
2 <= 3;
3 <= 2;
2.1 + 3.1;
2.1 - 3.1;
2.1 * 3.1;
4.2 / 2.1;
2.1 <= 3.1;
3.1 <= 2.1;
2 + 3.1;
2.1 - 3;
2 * 3.1;
2.1 / 3;
2 <= 3.1;
3.1 <= 2;
2 + 3 * 4 / 1 <= 5;
true;
false;
if true then 3 else 4;
if 4.6 <= 3 then if 2 <= .9 then 2 else 1 else 124.90;
0 / 0;
0 / 0.;
0. / 0;
0. / 0.;
NaN;
NaN + 2;
if NaN then 2 else 3;
4 +    -5   * 2;
let x = 2 + 3 in x <= 10;
let y = 10 in let y = 1 in y + y;
fun x -> x * 2 + 1;
(fun y -> if y then 1 else 0) (false);
(fun f -> f (10)) (fun var -> var -1);
let f1 = (fun x -> x + 1) in let f2 = (fun x -> x - 1) in f1(5) <= f2(5);
(fun x -> fun y -> x / y) (8) (2);
(fun x -> fun y -> x / y) (8);
