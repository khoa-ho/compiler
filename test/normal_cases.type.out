int
int
int
int
bool
bool
float
float
float
float
bool
bool
float
float
float
float
bool
bool
bool
bool
bool
int
float
int
int
float
float
float
int
bool
bool
bool
bool
int
(int -> int)
int
int
bool
float
(float -> float)
int
int
unit
(int * bool)
(bool * (float -> float))
int
bool
int
int
(bool -> (int * int))
(int * int)
int
[int]
[int]
bool
bool
int
<int>
int
unit
int
int
int
int
unit
array<int>
float
int
int
int
int
int
int
int
